## Github Repository for
# Genome-centric analyses of seasonal phyllosphere microbiome activities in perennial crops

## by Adina C. Howe, Nejc Stopnisek, Shane K. Dooley, Fan M. Yang, Keara L. Grady, and Ashley Shade


<i>This work is a preprint.</i>


### Data
The raw data (metagenomes and metatranscriptomes) for this study are available in the Joint Genomes Institute Genome Portal (https://genome.jgi.doe.gov/portal/ Project ID [503249](https://genome.jgi.doe.gov/portal/Seadynanfunction)) with projects designated by year and product type.  The MAG genomes analyzed in this paper are available on NCBI, as bioproject [PRJNA800073](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA800073).


### To cite this work or code
Howe AC, N Stopnisek, SK Dooley, FM Yang, KL Grady and A Shade.  2021.  Genome-centric analyses of seasonal phyllosphere microbiome activities in perennial crops.  bioRXiv. doi: https://doi.org/10.1101/2021.04.20.440608


### Abstract
Plants and microorganisms form beneficial associations. Understanding plant-microbe interactions will inform microbiome management to enhance crop productivity and resilience to stress. Here, we apply a genome-centric approach to identify key leaf microbiome members on field-grown switchgrass and miscanthus and quantify their activities for switchgrass over two growing seasons.  We integrate metagenome and metatranscriptome sequencing from 192 leaf samples collected over key time points in crop phenology. We curated 40 focal metagenome-assembled-genomes (MAGs) and conservatively focus analysis on transcript recruitment to medium and high-quality MAGs that were <10% contaminated and >50% complete. Classes represented by these MAGs (Actinomycetia, Alpha and Gamma- Proteobacteria, and Bacteroidota) were active and had seasonal dynamics in key functions, including enrichments in transcripts for of short chain dehydrogenase, molybdopterin oxioreductase, and polyketide cyclase in the late season. The majority of MAGs had activated stress-associated pathways, including trehalose metabolism, indole acetic acid degradation, betaine biosynthesis, and reactive oxygen species degradation, suggesting direct engagement with the host environment. We also detected seasonally activated biosynthetic pathways for terpenes (carotenoid and isoprenoids) and for various non-ribosomal peptide pathways that were poorly annotated. Overall, this study overcame laboratory and bioinformatic challenges associated with field-based leaf metatranscriptome analysis to inform potential key activities of these phyllosphere populations. These activities collectively support that leaf-associated bacterial populations are seasonally dynamic, responsive to host cues and interactively engage in feedbacks with the plant.  

### Organization of this repository
This analysis represented quality filtering of metagenomes and metatranscriptomes (data-preparation folder), metagenome assemblies (metagenome-assembly folder) and metagenome-assembled genome binning, curation, refinement, annotation (mag-evaluation folder).  Abundances of sequencing libraries were calculated based on reads mapped (mapping folder).  Additionallly, analysis of our annotated results are also included (analysis folder).  



### Funding
Support for this research was provided by the Great Lakes Bioenergy Research Center, U.S. Department of Energy, Office of Science, Office of Biological and Environmental Research (Awards DE-SC0018409 and DE-FC02-07ER64494), by the National Science Foundation Long-term Ecological Research Program (DEB 1637653 and 1832042) at the Kellogg Biological Station, by Michigan State University AgBioResearch, and by the DOE Center for Advanced Bioenergy and Bioproducts Innovation (US Department of Energy, Office of Science, Office of Biological and Environmental Research under award number DE-SC0018420).  This work was also supported in part by Michigan State University through computational resources provided by the Institute for Cyber-Enabled Research and in part by the University of Wisconsin-Madison Wisconsin Energy Institute as supported by GLBRC Information Services. The work conducted by the U.S. Department of Energy Joint Genome Institute, a DOE Office of Science User Facility, is supported under Contract No. DE-AC02-05CH11231.  AS acknowledges support by the USDA National Institute of Food and Agriculture and Michigan State University AgBioResearch. NS acknowledges support from the Michigan State University Plant Resilience Institute.  



### More info
[ShadeLab](http://ashley17061.wixsite.com/shadelab/home)

[GERMS Lab](http://www.germslab.org/about/)

[GLBRC](https://www.glbrc.org/)

[CABBI](https://cabbi.bio/)
